filebeat/input/input.go

```diff
@@ -48,12 +49,13 @@ type Input interface {
 
 // Runner encapsulate the lifecycle of the input
 type Runner struct {
-	config   inputConfig
-	input    Input
-	done     chan struct{}
-	wg       *sync.WaitGroup
-	Once     bool
-	beatDone chan struct{}
+	config         inputConfig
+	input          Input
+	done           chan struct{}
+	wg             *sync.WaitGroup
+	Once           bool
+	beatDone       chan struct{}
+	statusReporter status.StatusReporter
```

Comment:
Why do we need this field? I could not find where it's actually used. We have the status reported in the context and this is how it's propagated to the inputs, right?

Reply:
@rdner the situation get trickier, considering `log` input follows the old input interface https://github.com/elastic/beats/blob/1a6318d215517feb4c5c880619db8589d926956b/filebeat/input/input.go#L43-L47

We don't pass the `context` while executing `Run()` unlike the v2 interface.

With the v2 interface, the interface is 
https://github.com/elastic/beats/blob/1a6318d215517feb4c5c880619db8589d926956b/filebeat/input/v2/input.go#L54-L70

Reply:
> Why do we need this field? I could not find where it's actually used. We have the status reported in the context and this is how it's propagated to the inputs, right?

@rdner it is used here  
https://github.com/elastic/beats/blob/142ff3ec31e85cb3b91f3faaabad525383df1fd1/filebeat/input/input.go#L171-L173

It implements the `WithStatusReporter` interface and it's called here
https://github.com/elastic/beats/blob/d40a9e37285850a28ee95c474d4c9b3ac9e9e792/libbeat/cfgfile/list.go#L157-L161 
in a generic way for all the runners.